### PR TITLE
Dev/monthly reports mine

### PR DESCRIPTION
**Pull request by bakunyo**
https://github.com/hr-dash/hr-dash/pull/117 に依存
- マイ月報を追加
- 指定月の月報を登録できるよう修正

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -25,7 +25,11 @@
   end
   root to: 'sample#index'
 
-  resources 'monthly_reports', except: :destroy
+  resources 'monthly_reports', except: :destroy do
```
Comment (freezer530000) [inline config/routes.rb]: **[rails_best_practices]**
- restrict auto-generated routes monthly_reports (except: [:edit, :update, :destroy]) [:ref](http://rails-bestpractices.com/posts/2011/08/19/restrict-auto-generated-routes/)

File: app/controllers/monthly_reports_controller.rb [hunk 2]
```diff
@@ -1,6 +1,20 @@
 class MonthlyReportsController < ApplicationController
   def index
-    @monthly_reports = MonthlyReport.all
+    @monthly_reports = MonthlyReport.page params[:page]
+  end
+
+  def mine
```
Comment (freezer530000) [inline app/controllers/monthly_reports_controller.rb]: **[rubocop]**
- Assignment Branch Condition size for mine is too high. [22.93/15] [:ref](http://c2.com/cgi/wiki?AbcMetric)

File: app/models/monthly_report.rb [hunk 3]
```diff
@@ -25,13 +25,40 @@ class MonthlyReport < ActiveRecord::Base
   validates :user_id, numericality: { only_integer: true }, presence: true
   validates :target_month, presence: true
   validate :target_beginning_of_month?
+  validate :target_month_registrable_term
+
+  scope :year, -> year { where(target_month: (Time.zone.local(year))..(Time.zone.local(year).end_of_year)) }
```
Comment (freezer530000) [inline app/models/monthly_report.rb]: **[rubocop]**
- Wrap stabby lambda arguments with parentheses. [:ref](https://github.com/bbatsov/ruby-style-guide#stabby-lambda-with-args)

File: app/controllers/monthly_reports_controller.rb (hunk 2 above)
Comment (bakunyo) [inline app/controllers/monthly_reports_controller.rb]: e984db3 で解消済（のはず）

File: config/routes.rb (hunk 1 above)
Comment (bakunyo) [inline config/routes.rb]: edit, updateは今後つくるのでこれでOK

**Comment by tamata78** [comment]
LGTM
LGTM

File: config/routes.rb [hunk 4]
```diff
@@ -26,8 +26,12 @@
 
   root to: 'sample#index'
 
-  resources 'monthly_reports', except: :destroy
   resources 'user_profiles', except: [:index, :show, :destroy]
+  resources 'monthly_reports', except: :destroy do
```
Comment (freezer530000) [inline config/routes.rb:30]: **[rails_best_practices]**
- restrict auto-generated routes monthly_reports (except: [:edit, :update, :destroy]) [:ref](http://rails-bestpractices.com/posts/2011/08/19/restrict-auto-generated-routes/)

**Comment by ta-chibana** [comment]
LGTM

**Comment by Nobuhiko-Kudo** [comment]
:ok_woman:

**Pull request merged**